node_modules/oae-core/documentpreview/js/documentpreview.js

```diff
@@ -71,6 +71,93 @@ define(['jquery', 'underscore', 'oae.core', 'lazyload'], function($, _, oae) {
         ////////////////////////////////
 
         /**
+         * Prepare markup for a page to be loaded
+         *
+         * @param  {Object}         page                Page object representing the page that will be loaded
+         */
+        var prepareForPageLoading = function(page) {
+            // Add the page container, containing a loading indicator
+            $spacer.before(oae.api.util.template().render($('#documentpreview-content-page-template', $rootel), {
+                'cssScopeClass': widgetData.previews.cssScopeClass,
+                'pageNumber': page.pageNumber
+            }));
+
+            // Cache a reference to the page element
+            page.$el = $('.documentpreview-content-page[data-page-number="' + page.pageNumber + '"]', $rootel);
+
+            // Cache the height and calculated top margin of the
+            // element as they are needed for zoom calculations
+            page.height = page.$el.height();
+            page.marginTop = CONTENT_PAGE_SPACING;
+        };
+
+        /**
+         * Render the HTML content for a page
+         *
+         * @param  {Object}         page                Page object representing the page that will be loaded
+         * @param  {String}         html                HTML content for the page
+         */
+        var renderPageHtml = function(page, html) {
+            // Remove the loading indicator
+            $('.documentpreview-content-page-loading', page.$el).remove();
+
+            // Add the HTML content
+            page.$html = $('.documentpreview-content-html-page', page.$el).html(html);
+
+            // Update the height now that we have real content
+            page.height = page.$el.height();
+
+            // If the page being loaded is the first page of the document, we adjust
+            // the zoom level based on the available space to make sure that the page
+            // fully fits inside of the viewer
+            if (page.pageNumber === 1) {
+                determineZoomLevel(page);
+            }
+
+            // Adjust the page display and any pages following it to reflect the
+            // current zoom level
+            zoomPages(page.pageNumber);
+        };
+
+        /**
+         * Render the text content for a page (for screen readers)
+         *
+         * @param  {Object}         page                Page object representing the page that will be loaded
+         * @param  {String}         text                Text content for the page
+         */
+        var renderPageText = function(page, text) {
```

Comment:
I would fetch the text content as part of this function, and pass in a callback to call when the text page has been loaded and rendered